Document epp:scan_erl_form/1 and add epp:scan_file/2

richcarl opened the pull request:
epp:scan_erl_form/1 was undocumented, but in use by edoc and there were tests for it.
epp:scan_file/2 can be used to implement e.g. a standalone token level preprocessor pass, like gcc -E.

jhogberg commented on lib/stdlib/src/epp.erl:
```diff
@@ -232,6 +237,39 @@ format_error({warning,Term}) ->
+		  {'source_name', SourceName :: file:name()} |
+		  {'macros', PredefMacros :: macros()} |
+		  {'default_encoding', DefEncoding :: source_encoding()}],
+      Form :: erl_scan:tokens() | {'error', ErrorInfo} | {'eof', Loc},
```
dialyzer is unhappy over `erl_scan:tokens/0` not being exported. Can you export it? I figure it's just an oversight since it's documented and the closely related `erl_scan:token/0` is exported.

jhogberg commented on lib/stdlib/src/epp.erl:
```diff
@@ -232,6 +237,39 @@ format_error({warning,Term}) ->
+
+scan_file(Epp) ->
+    case scan_erl_form(Epp) of
+	{ok,Toks} ->
```
`dialyzer` is still unhappy :(

```
epp.erl:265: The pattern 
          {'ok', Toks} can never match the type 
          {'eof',
           non_neg_integer() | {non_neg_integer(), pos_integer()}} |
          {'error',
           {non_neg_integer() | {non_neg_integer(), pos_integer()},
            atom(),
            _},
           non_neg_integer() | {non_neg_integer(), pos_integer()}} |
          {'ok',
           [{atom(), erl_anno:anno()} |
            {atom(), erl_anno:anno(), atom() | string() | number()}],
           non_neg_integer() | {non_neg_integer(), pos_integer()}}
epp.erl:267: The pattern 
          {'error', E} can never match the type 
          {'eof',
           non_neg_integer() | {non_neg_integer(), pos_integer()}} |
          {'error',
           {non_neg_integer() | {non_neg_integer(), pos_integer()},
            atom(),
            _},
           non_neg_integer() | {non_neg_integer(), pos_integer()}} |
          {'ok',
           [{atom(), erl_anno:anno()} |
            {atom(), erl_anno:anno(), atom() | string() | number()}],
           non_neg_integer() | {non_neg_integer(), pos_integer()}}
```

richcarl commented:
Fixed type spec, rebased and squashed. But I can't see why it's failing now: `Missing Anchor: "/lib/stdlib-3.14/doc/html/epp.html#type-warning_info" from /lib/stdlib-3.14/doc/html/epp.html` - the warning_info() type was already used in the spec of parse_erl_form/1 which was already exported - cf. http://erlang.org/doc/man/epp.html#parse_erl_form-1

garazdawi commented:
> Fixed type spec, rebased and squashed. But I can't see why it's failing now: `Missing Anchor: "/lib/stdlib-3.14/doc/html/epp.html#type-warning_info" from /lib/stdlib-3.14/doc/html/epp.html` - the warning_info() type was already used in the spec of parse_erl_form/1 which was already exported - cf. http://erlang.org/doc/man/epp.html#parse_erl_form-1

You need to either put the `warning_info` type in the `<datatypes>` section or explicitly inline the type into the function docs as [parse_erl_form/1](https://github.com/erlang/otp/blob/OTP-23.2/lib/stdlib/doc/src/epp.xml#L160) has done.

richcarl commented:
Fixed doc and rebased.

jhogberg commented:
Merged, thanks for the PR!
